### ISSUE TITLE
Helm v3 binary in Pulumi actions docker image possibly breaks helm fetches on GitHub actions

### ISSUE THREAD

**Issue by kahkeng** [pulumi/pulumi#4397]
I'm getting the following error when using the latest Pulumi actions docker image for GitHub actions to deploy a helm chart.

```
2020-04-15T00:15:19.5419814Z Diagnostics:
2020-04-15T00:15:19.5420479Z   pulumi:pulumi:Stack (clusterServices-utils):
2020-04-15T00:15:19.5421018Z     error: Program failed with an unhandled exception:
2020-04-15T00:15:19.5421552Z     error: Traceback (most recent call last):
2020-04-15T00:15:19.5422423Z       File "/usr/bin/pulumi-language-python-exec", line 85, in <module>
2020-04-15T00:15:19.5423018Z         loop.run_until_complete(coro)
2020-04-15T00:15:19.5423601Z       File "/usr/local/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
2020-04-15T00:15:19.5424173Z         return future.result()
2020-04-15T00:15:19.5424894Z       File "/usr/local/lib/python3.7/site-packages/pulumi/runtime/stack.py", line 81, in run_in_stack
2020-04-15T00:15:19.5425502Z         await run_pulumi_func(lambda: Stack(func))
2020-04-15T00:15:19.5426252Z       File "/usr/local/lib/python3.7/site-packages/pulumi/runtime/stack.py", line 50, in run_pulumi_func
2020-04-15T00:15:19.5426850Z         await RPC_MANAGER.rpcs.pop()
2020-04-15T00:15:19.5427606Z       File "/usr/local/lib/python3.7/site-packages/pulumi/runtime/rpc_manager.py", line 67, in rpc_wrapper
2020-04-15T00:15:19.5428188Z         result = await rpc
2020-04-15T00:15:19.5428922Z       File "/usr/local/lib/python3.7/site-packages/pulumi/runtime/resource.py", line 440, in do_register_resource_outputs
2020-04-15T00:15:19.5429560Z         serialized_props = await rpc.serialize_properties(outputs, {})
2020-04-15T00:15:19.5430355Z       File "/usr/local/lib/python3.7/site-packages/pulumi/runtime/rpc.py", line 68, in serialize_properties
2020-04-15T00:15:19.5431151Z         result = await serialize_property(v, deps, input_transformer)
2020-04-15T00:15:19.5432064Z       File "/usr/local/lib/python3.7/site-packages/pulumi/runtime/rpc.py", line 173, in serialize_property
2020-04-15T00:15:19.5433311Z         value = await serialize_property(output.future(), deps, input_transformer)
2020-04-15T00:15:19.5434624Z       File "/usr/local/lib/python3.7/site-packages/pulumi/runtime/rpc.py", line 159, in serialize_property
2020-04-15T00:15:19.5435457Z         future_return = await asyncio.ensure_future(awaitable)
2020-04-15T00:15:19.5436506Z       File "/usr/local/lib/python3.7/site-packages/pulumi/output.py", line 114, in get_value
2020-04-15T00:15:19.5437304Z         val = await self._future
2020-04-15T00:15:19.5438266Z       File "/usr/local/lib/python3.7/site-packages/pulumi/output.py", line 155, in run
2020-04-15T00:15:19.5439079Z         value = await self._future
2020-04-15T00:15:19.5440040Z       File "/usr/local/lib/python3.7/site-packages/pulumi/output.py", line 155, in run
2020-04-15T00:15:19.5440820Z         value = await self._future
2020-04-15T00:15:19.5441793Z       File "/usr/local/lib/python3.7/site-packages/pulumi/output.py", line 155, in run
2020-04-15T00:15:19.5442575Z         value = await self._future
2020-04-15T00:15:19.5443271Z       [Previous line repeated 3 more times]
2020-04-15T00:15:19.5444263Z       File "/usr/local/lib/python3.7/site-packages/pulumi/output.py", line 186, in run
2020-04-15T00:15:19.5445064Z         return await transformed.future(with_unknowns=True)
2020-04-15T00:15:19.5446107Z       File "/usr/local/lib/python3.7/site-packages/pulumi/output.py", line 114, in get_value
2020-04-15T00:15:19.5446913Z         val = await self._future
2020-04-15T00:15:19.5447870Z       File "/usr/local/lib/python3.7/site-packages/pulumi/output.py", line 155, in run
2020-04-15T00:15:19.5448673Z         value = await self._future
2020-04-15T00:15:19.5463192Z       File "/usr/local/lib/python3.7/site-packages/pulumi/output.py", line 176, in run
2020-04-15T00:15:19.5464048Z         transformed: Input[U] = func(value)
2020-04-15T00:15:19.5465146Z       File "/usr/local/lib/python3.7/site-packages/pulumi_kubernetes/helm/v2/helm.py", line 325, in _parse_chart
2020-04-15T00:15:19.5466062Z         _fetch(chart_to_fetch, fetch_opts)
2020-04-15T00:15:19.5467156Z       File "/usr/local/lib/python3.7/site-packages/pulumi_kubernetes/helm/v2/helm.py", line 407, in _fetch
2020-04-15T00:15:19.5468148Z         subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, check=True)
2020-04-15T00:15:19.5469024Z       File "/usr/local/lib/python3.7/subprocess.py", line 512, in run
2020-04-15T00:15:19.5469809Z         output=stdout, stderr=stderr)
2020-04-15T00:15:19.5471305Z     subprocess.CalledProcessError: Command '['helm', 'fetch', 'stable/wordpress', '--untar', '--version', '2.1.3', '--destination', '/tmp/tmp9q63lhrf']' returned non-zero exit status 1.
2020-04-15T00:15:19.5472422Z     error: an unhandled error occurred: Program exited with non-zero exit code: 1
```

I'm using pulumi 1.14.0 and pulumi_kubernetes 1.6.0 on python SDK.

This script to install a Helm chart modified from https://www.pulumi.com/docs/guides/adopting/from_kubernetes/ to use helm.v3 is what gave the above error. (Although I do see that v2 and v3 are identical right now in pulumi-kubernetes.)

```
import pulumi
from pulumi_kubernetes.helm.v3 import Chart, ChartOpts

# Deploy the latest version of the stable/wordpress chart.
wordpress = Chart('wpdev', config=ChartOpts(
    repo='stable',
    chart='wordpress',
    version='2.1.3',
))
```

With some debugging, it seems like the error that happens with the command is `Error: repo stable not found`.

Here is my guess at what is happening. It seems like with Helm v3, according to https://helm.sh/docs/faq/#xdg-base-directory-support, the repo configs can be overridden with these environment variables, but otherwise they default to the following:
```
XDG_CONFIG_HOME=$HOME/.config
XDG_CACHE_HOME=$HOME/.cache
```

And on GitHub actions, HOME environment defaults to `/github/home` so it is unable to find the stable repo that was added in https://github.com/pulumi/pulumi/pull/4090/files#diff-2af82e9e4443a83f71216d2624bcac08R59.

If I added these lines to my GitHub workflow yaml, then the above error goes away:
```
        env:
          XDG_CONFIG_HOME: /root/.config
          XDG_CACHE_HOME: /root/.cache
```

But a more general fix might be to explicitly add these lines to either the base Pulumi docker image or the Pulumi actions docker image so that things could work out of the box without people having to do these on their own configs:
```
ENV XDG_CONFIG_HOME=/root/.config
ENV XDG_CACHE_HOME=/root/.cache
```

I can submit a PR for this if it makes sense.

**Comment by lblackstone** [comment]
Thanks for the report @kahkeng -- I fixed the linked issue with the `HELM_HOME` bug. Would be happy to review a PR for this issue. Thanks!

**Comment by kahkeng** [comment]
@lblackstone I just made a PR for this. I chose to put this in the base pulumi docker image instead of the actions one because I felt it would be a more general fix.

**Comment by stack72** [comment]
Closed via #4397
